src/index.ts

```diff
@@ -488,9 +604,240 @@ export const readable = <T>(
+    }
+    set(value);
+  };
+
+  const synchronize = async (set: Subscriber<T>): Promise<T> => {
+    const storageKey = await getKey();
+    const storageItem = getStorageItem(storageKey);
+
+    if (storageItem) {
+      const stored = JSON.parse(storageItem);
+      set(stored);
+
+      return stored;
+    } else if (initial !== undefined) {
+      if (isLoadable(initial)) {
+        const $initial = await initial.load();
+        await setAndPersist($initial, set);
+
+        return $initial;
+      } else {
+        await setAndPersist(initial, set);
+
+        return initial;
+      }
+    } else {
+      set(undefined);
+      return undefined;
+    }
+  };
+
+  let initialSync: Promise<T>;
+
+  const thisStore = writable<T>(undefined, (set) => {
+    initialSync = synchronize(set);
+  });
+
+  const subscribe = thisStore.subscribe;
+
+  const set = async (value: T) => {
+    await initialSync;
+    return setAndPersist(value, thisStore.set);
+  };
+
+  const update = async (updater: Updater<T>) => {
+    await (initialSync ?? synchronize(thisStore.set));
+    const newValue = updater(get(thisStore));
+    await setAndPersist(newValue, thisStore.set);
+  };
+
+  const load = thisStore.load;
+
+  const reload = reloadable
+    ? async () => {
+        let newValue: T;
+
+        if (isLoadable(initial)) {
+          [newValue] = await reloadAll([initial]);
+        } else {
+          newValue = initial;
+        }
+
+        setAndPersist(newValue, thisStore.set);
+        return newValue;
+      }
+    : undefined;
+
+  const resync = async (): Promise<T> => {
+    await initialSync;
+    return synchronize(thisStore.set);
+  };
+
+  return {
+    get store() {
+      return this;
+    },
+    subscribe,
+    set,
+    update,
+    load,
+    resync,
+    ...(reload && { reload }),
+  };
+};
+
+// ASYNC CLIENT
+
+type AsyncClient<T> = T extends (...args: infer TArgs) => infer TReturn
+  ? (...args: TArgs) => Promise<TReturn>
+  : {
+      [k in keyof T]: T[k] extends (...args: infer KArgs) => infer KReturn // callable property?
+        ? (...args: KArgs) => Promise<KReturn> // make the function async
+        : () => Promise<T[k]>; // return the property in a Promise
+    };
+
+/**
+ * Generates an AsyncClient from a Loadable store. The AsyncClient will have all
+ * of the properties of the input store, plus a collection of asynchronous functions
+ * for kicking off access of the store's value's properties before it has finished loading.
+ * i.e. an asyncClient that loads to {foo: 'bar'} will have a `foo` function that
+ * resolves to 'bar' when the store has loaded.
+ * @param loadable Loadable to unpack into an asnycClient
+ * @returns an asyncClient with the properties of the input store and asynchronous
+ * accessors to the properties of the store's loaded value
+ */
+export const asyncClient = <S extends Loadable<unknown>>(
+  loadable: S
+): S & AsyncClient<StoresValues<S>> => {
+  return new Proxy(Function.prototype, {
```

Comment:
This is causing problems with jest spyOn

We proxy the Function.prototype here so that we can invoke asyncClients if they represent a function.

i.e. 
```
const myClient = asyncClient(
  derived(
    someLoadable, 
    ($someLoadable) => {
      return (val) => $someLoadable + val
    } 
  )
);
// the store loads to a function, so we can invoke an asyncClient
myClient(2);
```

 However this means that all asyncClients are considered to be functions, even if they represent an object with properties. This means jest doesn't understand if you try to spy on any properties of the asyncClient. So if you have an asyncClient which loads to be 
 ```
 { someFunc: () => 'foo' }
 ```
 Attempting to spy on someFunc will fail `jest.spyOn(myClient, 'someFunc');` 
 "Matcher error: received value must be a mock or spy function"